### ISSUE TITLE
Maccy not appear in screen center sometimes on macOS 14.0 (23A344)

### ISSUE THREAD

**Issue by rea1shane** [p0deje/Maccy#612]
Sometimes it appear in top center. When I press `cmd` + `shift` + `C` again, a beep will appear.

Maccy version 0.27.0

**Comment by samtsevich** [comment]
That depends on the active window when you try to run Maccy.
For example, in Safari, Cmd+Shift+C is a shortcut for "Start element selection" in Developer Mode.
Thus, I just changed the shortcut that runs Maccy to Cmd-Option-C.

**Comment by p0deje** [comment]
2. Can you please record a screen video showing the problem? You can do this using QuickTime.app.
3. Can you please share the output of running `defaults read org.p0deje.Maccy` in Terminal.app?

**Comment by weisJ** [comment]
I have had this happen as well. I think it occurs if a maximum number of history is specified. Somehow too many items (probably all) are added to the menu causing it to have maximum size (hence the popup being pushed up to the top). Because the item count is limited immediately before the popup is shown one doesn’t see that the menu was actually too large at time of opening.

**Comment by rea1shane** [comment]
![录屏2023-09-28 10 37 15](https://github.com/p0deje/Maccy/assets/93363751/a3f43da5-f53c-4845-927b-33cda67c81a4)

When I press `CMD` + `Shift` + `C` to close the window quickly, there'll be a bell.

Rarely it appear correctly in the center.

```yaml
{
    "KeyboardShortcuts_popup" = "{\\"carbonModifiers\\":768,\\"carbonKeyCode\\":8}";
    "LaunchAtLogin__hasMigrated" = 1;
    "NSStatusItem Preferred Position Item-0" = 5675;
    "NSWindow Frame SUStatusFrame" = "700 743 400 134 0 0 1800 1125 ";
    "NSWindow Frame SUUpdateAlert" = "590 545 620 398 0 0 1800 1125 ";
    "NSWindow Frame com.sindresorhus.Preferences.FrameAutosaveName" = "679 150 383 646 0 0 1920 1055 ";
    SUAutomaticallyUpdate = 1;
    SUEnableAutomaticChecks = 1;
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2023-09-27 06:59:21 +0000";
    SUUpdateGroupIdentifier = 3766769327;
    WebKitDefaultFontSize = 13;
    WebKitJavaEnabled = 0;
    WebKitJavaScriptEnabled = 0;
    WebKitPluginsEnabled = 0;
    WebKitStandardFont = "-apple-system-font";
    WebKitUserStyleSheetEnabledPreferenceKey = 1;
    WebKitUserStyleSheetLocationPreferenceKey = "/Applications/Maccy.app/Contents/Frameworks/Sparkle.framework/Resources/ReleaseNotesColorStyle.css";
    enabledPasteboardTypes =     (
        "public.html",
        "public.utf8-plain-text",
        "public.tiff",
        "public.rtf",
        "public.file-url",
        "public.png"
    );
    hideFooter = 1;
    hideSearch = 0;
    hideTitle = 1;
    history = {length = 42, bytes = 0x62706c69 73743030 a0080000 00000000 ... 00000000 00000009 };
    historySize = 999;
    ignoredPasteboardTypes =     (
        "com.typeit4me.clipping",
        "com.agilebits.onepassword",
        "Pasteboard generator type",
        "net.antelle.keeweb",
        "de.petermaurer.TransientPasteboardType"
    );
    maxMenuItems = 20;
    menuIcon = clipboard;
    migrations =     {
        "2020-02-22-history-item-add-copied-at" = 1;
        "2020-02-22-history-item-add-number-of-copies" = 1;
        "2020-04-18-switch-storage-to-core-data" = 1;
        "2020-04-25-allow-custom-ignored-types" = 1;
        "2020-06-19-use-keyboardshortcuts" = 1;
        "2020-09-01-ignore-keeweb" = 1;
        "2021-02-20-allow-to-customize-supported-types" = 1;
        "2021-06-28-add-title-to-history-item" = 1;
        "2021-10-16-remove-dynamic-pasteboard-types" = 1;
        "2022-08-01-rename-suppress-clear-alert" = 1;
        "2022-11-14-add-html-rtf-to-supported-types" = 1;
        "2023-01-22-add-regexp-search-mode" = 1;
    };
    pasteByDefault = 0;
    pinTo = top;
    playSounds = 0;
    popupPosition = center;
    popupScreen = 0;
    removeFormattingByDefault = 0;
    showRecentCopyInMenuBar = 0;
}
```

**Comment by weisJ** [comment]
I can replicate this reliably (every time) in Sonoma. The menu does indeed report the a too large size before being opened.
However I haven't quite figured out the changes necessary so it reports the correct size. Once it is open it has
the `frame` size is correct but doesn't survive until next opened.

**Comment by weisJ** [comment]
It appears that the only solution is to not add hidden items to the menu. I am currently exploring how feasible that is.

**Comment by p0deje** [comment]
Fixed in #631
